apis/pkg/meta/register.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package meta
+
+import (
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+// Package type metadata.
+const (
+	Group   = "meta.pkg.crossplane.io"
+	Version = "meta"
+)
```

Comment:
I have a hunch that it could make our lives easier as we get deeper into conversions. I like that it would give us fixed types to build our actual code around. Versioned 'user facing' types could be simple types with no logic and our getters and setters, composition methods, etc could be associated with only the internal type. We'd also never need to go through the process of bumping our controllers to a wholesale new type (e.g. `s/v1beta1/v1/g` on the codebase). I suspect there could be some more advantages for roundtrippability such as those laid out in ASO's [research on this](https://github.com/Azure/k8s-infra/blob/611763e4ae99501b69372535020774ed46529125/hack/generator/spec/versioning.md).

That all said, I agree that it's not the right time to introduce such a pattern so I've pushed a commit that removes the internal types and uses `v1` as the conversion hub.